src/models/matches.rs

```diff
@@ -80,6 +118,113 @@ impl Match {
       },
     }
   }
+
+  // Populates the leading and trailing comma and comment ranges for the match.
+  fn populate_associated_elements(
+    &mut self, node: &Node, code: &String, piranha_arguments: &PiranhaArguments,
+  ) {
+    self.get_associated_elements(node, code, piranha_arguments, true);
+    self.get_associated_elements(node, code, piranha_arguments, false);
+  }
+
+  /// Get the associated elements for the match.
+  /// We currently capture leading and trailing comments and commas.
+  fn get_associated_elements(
+    &mut self, node: &Node, code: &String, piranha_arguments: &PiranhaArguments, trailing: bool,
+  ) {
+    let mut current_node = *node;
+    let mut buf = *piranha_arguments.cleanup_comments_buffer();
+    let mut found_comment = !self.associated_comments().is_empty();
+    let mut found_comma = self.associated_comma().is_some();
+    loop {
+      // If we are looking for trailing elements, we start from the next sibling of the node
+      // Else we start from the previous sibling of the node
+      while let Some(sibling) = if trailing {
+        current_node.next_sibling()
+      } else {
+        current_node.prev_sibling()
+      } {
+        let content = sibling.utf8_text(code.as_bytes()).unwrap();
+        if !found_comma && content.trim().eq(",") {
```

Comment:
Question: If we are removing leading and trailing commas, wouldn't deleting `b` in `foo(a, b, c)` lead to broken code like `foo(a c)` or are we relying on the whitespace to prevent that? And, if the later, what's an example of a comma before the element that we _do_ want to remove?

Reply:
> Similarly if we find trailing comments we do not lookup leading comments.

Is this true? Or even what we want? I think we keep looking for comments even if we already found some. The only case in which we look at `found_comment` is to decide whether or not to expand our search to the parent node's siblings, and we still do that if we have found comments but not a comma. No?

This also seems to be reasonable, because we would expect both comments to be deleted in:

```
// The user type is used for [bla bla]
private UserType userType = DEFAULT_USER_TYPE; // DEFAULT_USER_TYPE == 0
```

Reply:
I stand corrected. 
If we find a trailing comment, we will still try to lookup a leading comment. 
Yes, found comment guards our `parent-level` search. 

I have updated our test cases to show that the current implementation handles the scenario you mention above: 

https://github.com/uber/piranha/pull/410/commits/d1c301c9cdc19b4f78a2ca55a2d8fc34b3216611
^ I added a comment on the line of enum (Which got cleaned up, hence the expected was not updated)